dask/config.py

```diff
@@ -434,4 +436,26 @@ def update_defaults(new, config=config, defaults=defaults):
     update(config, new, priority='old')
 
 
+def expand_environment_variables(config):
+    ''' Utility to expand environment variables in a config dictionary
+
+    This function will recursively search through any nested dictionaries
+    and/or lists.
+
+    config : dict, iterable, or str
+        Input object to search for environment variables
```

Comment:
I think that this should go in a `Parameters` section.

We might consider adding a small examples section as well like the following:

```python
Examples
--------
>>> expand_environment_variables({'x': [1, 2, '$USER']})
{'x': [1, 2, 'my-username']}
```